plugins/woocommerce/src/Admin/Features/Blueprint/RestApi.php

```diff
@@ -0,0 +1,178 @@
+ *
+ * @package Automattic\WooCommerce\Admin\Features\Blueprint
+ */
+class RestApi {
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'blueprint';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'process';
+
+	/**
+	 * Register routes.
+	 *
+	 * @since 3.5.0
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'import' ),
+					'permission_callback' => function () {
+						return true;
+					},
+				),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/export',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'export' ),
+					'permission_callback' => function () {
+						return true;
+					},
+					'args'                => array(
+						'steps'         => array(
+							'description'       => 'A list of plugins to install',
+							'type'              => 'array',
+							'items'             => 'string',
+							'default'           => array(),
+							'sanitize_callback' => function ( $value ) {
+								return array_map(
+									function ( $value ) {
+										return sanitize_text_field( $value );
+									},
+									$value
+								);
+							},
+							'required'          => false,
+						),
+						'export_as_zip' => array(
+							'description' => 'Export as a zip file',
+							'type'        => 'boolean',
+							'default'     => false,
+							'required'    => false,
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Handle the export request.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 * @return \WP_HTTP_Response The response object.
+	 */
+	public function export( $request ) {
+		$steps         = $request->get_param( 'steps', array() );
+		$export_as_zip = $request->get_param( 'export_as_zip', false );
+		$exporter      = new ExportSchema();
+
+		$data = $exporter->export( $steps, $export_as_zip );
+
+		if ( $export_as_zip ) {
+			$zip  = new ZipExportedSchema( $data );
+			$data = $zip->zip();
+			$data = site_url( str_replace( ABSPATH, '', $data ) );
+		}
+
+		return new \WP_HTTP_Response(
+			array(
+				'data' => $data,
+				'type' => $export_as_zip ? 'zip' : 'json',
+			)
+		);
+	}
+
+	/**
+	 * Handle the import request.
+	 *
+	 * @return \WP_HTTP_Response The response object.
+	 */
+	public function import() {
+		// phpcs:ignore
+		if ( ! empty( $_FILES['file'] ) && $_FILES['file']['error'] === UPLOAD_ERR_OK ) {
+			// phpcs:ignore
+			$uploaded_file = $_FILES['file']['tmp_name'];
+
+			try {
+				// phpcs:ignore
+				if ( $_FILES['file']['type'] === 'application/zip' ) {
+					// phpcs:ignore
+					$newpath = \wp_upload_dir()['path'] . '/' . $_FILES['file']['name'];
```

Comment:
@rjchow Good catch! Changed it to use the built-in [wp_handle_upload](https://developer.wordpress.org/reference/functions/wp_handle_upload/) function in [Use built-in wp_handle_upload to avoid any potential known security i…](https://github.com/woocommerce/woocommerce/pull/49257/commits/7a5e55c80c3762478a658ca1993aade9f5ed37cd)